src/vs/platform/windows/electron-main/window.ts

```diff
@@ -441,16 +443,45 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 			return false;
 		};
 
-		const isRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
+		const isSvgRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
 			return details.resourceType === 'xhr' || isSafeFrame(details.frame);
 		};
 
+		const workbenchRootUrl = FileAccess.asBrowserUri('vs/', require).toString(true);
+		const isAllowedVsCodeFileRequest = (uri: URI, details: Electron.OnBeforeRequestListenerDetails) => {
+			// Allow vscode-file requests from main window
+			if (details.frame?.url.startsWith('vscode-file://vscode-app/')) {
+				return true;
+			}
+
+			if (details.frame?.url || details.frame?.parent) { // Coming from unrecognized frame
+				return false;
+			}
+
+			// Dealing with frame without a url set yet. This happens when first loading a window.
+			// Here we still want to allow loading the main workbench html files
+			if (details.url.startsWith(workbenchRootUrl) && uri.path.endsWith('.html')) {
+				return true;
+			}
+
+			return false;
+		};
+
 		this._win.webContents.session.webRequest.onBeforeRequest((details, callback) => {
 			const uri = URI.parse(details.url);
+
+			if (uri.scheme === Schemas.vscodeFileResource) {
+				if (!isAllowedVsCodeFileRequest(uri, details)) {
+					console.log('Blocked vscode-file request', details.url);
```

Comment:
Should this use our logger rather with `logService.error` similar to our protocol handling: https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/protocol/electron-main/protocolMainService.ts#L77?